pkg/name/registry.go

```diff
@@ -50,6 +51,12 @@ func (r Registry) String() string {
 	return r.Name()
 }
 
+// Repo returns a Repository in the Registry with the given name.
+func (r Registry) Repo(repo ...string) Repository {
+	// TODO: Panic if repo is empty?
```

Comment:
This is also kind of a wart. What should we do if someone does `NewRegistry("foo.io").Repo()`? Now they have a `name.Repository` with an empty repository, which they can pass to `remote.List` and send a request to `https://foo.io/v2/tags/list` where it will fail. Same for `remote.WriteLayer` and `https://foo.io/v2/blobs/uploads`.

Should we just panic here in that case?

Reply:
> where it will fail

It depends on the registry. Some registries will not fail here, which is ~most of the point of adding this.

Reply:
Okay cool. I won't panic, but I will look at it funny if I ever see it. :)